test/CMakeLists.txt

```diff
@@ -40,13 +40,6 @@ macro(onedpl_add_test test_source_file split_kernels)
         target_compile_options(${_test_name} PRIVATE $<$<CONFIG:Release>:-fno-strict-aliasing>)
     endif()
 
-    # oneDPL tests of std::complex required generates code to do JIT compilation of a kernel only when it is called
-    if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang|IntelLLVM" )
-        if (${split_kernels})
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsycl-device-code-split=per_kernel ")
-        endif()
-    endif()
-
```

Comment:
Would not it be enough to only remove (or comment out) this part, while keeping the other #691 changes in this file? Or do you think the make/cmake changes all need to be done in a very different way?

Reply:
I think this section must be reworked too. It enables`-fsycl-device-code-split=per_kernel` option for all the tests (verified practically) with intention to use it for just `xpu_api/numerics/complex.number`, which is not a full solution either: we have other tests using double types.

Reply:
I suggest reverting the changes in cmake/make to think about a better solution. There are lot of issues with these changes, while they are not essential.

Reply:
I suggest using compiler option check, which is supported by cmake, like this:
`check_cxx_compiler_flag("-fsycl" _fsycl_option)`

Reply:
I am working on these changes in #704 to be able to select #702 or #704.

Reply:
@SergeyKopienko, let me apologize, I analyzed the issues and found that #691 has a good code base that can be reused with some additional fixes. #704 is inspired by the changes done in #691.

Reply:
Fine, NP, thanks!